### PR TITLE
[JENKINS-46541] Do not use expired login sessions - simple version

### PR DESCRIPTION
**Pull request by pjdarton**
Fixes [JENKINS-46541](https://issues.jenkins-ci.org/browse/JENKINS-46541) as discussed in [comment](https://issues.jenkins-ci.org/browse/JENKINS-46541?focusedCommentId=315197&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-315197).
JCloudsCloud no longer caches Openstack instances - that's the Openstack factory class' responsibility.
Openstack no longer tracks when its login token is due to expire, it simply caches each login for a fixed period (10 minutes).

### CODE REVIEW

**Comment by stpierre** [comment]
Yes please. This fixes a very painful issue. :+1:

**Comment by signed8bit** [comment]
Ditto. We'd love to see this merged and released.

File: src/main/java/jenkins/plugins/openstack/compute/JCloudsCloud.java [hunk 1]
```diff
@@ -78,9 +78,6 @@
     private transient @Deprecated Integer startTimeout;
     private transient @Deprecated Boolean floatingIps;
 
-    // Cache the instance between uses
-    private volatile transient @CheckForNull Openstack openstack;
```
Comment (olivergondza) [inline src/main/java/jenkins/plugins/openstack/compute/JCloudsCloud.java]: My intention here was to ensure that threads that repeatedly talk to the same cloud at about the same time (AJAX calls for form validation) would reuse the same instance of {{Openstack}} that will take care of creating dedicated client instance based on existing token. IOW, one auth dance per instance. This has the obvious flaw of not expiring the instance.

As I understand the change, the logic was moved but the behavior is the same plus it invalidates with fixed time.

Comment (pjdarton) [reply src/main/java/jenkins/plugins/openstack/compute/JCloudsCloud.java]: Correct - the (old) functionality in JCloudsCloud meant that the Openstack instance was never expired.
This PR keeps the same same error handling but now every call to `JCloudsCloud.getOpenstack()` delegates to `Openstack.Factory.get(...)` and Openstack.Factory provides both caching _and_ token expiry.

i.e. the old code did caching without expiry, the new code provides caching with expiry.

**Comment by olivergondza** [comment]
Merging slightly amended: https://github.com/jenkinsci/openstack-cloud-plugin/commit/f3701adca9babd90cc5a14f7ea66d84b65061596#diff-156d9e9d86305aa1ba97f70a959e06c6L789. Thanks!

**Pull request merged**